iotdb-core/datanode/src/main/java/org/apache/iotdb/db/audit/DNAuditLogger.java

```diff
@@ -0,0 +1,304 @@
+              TSFileConfig.STRING_CHARSET),
+          auditLogFields.isResult(),
+          new Binary(
+              auditLogFields.getDatabase() == null ? "null" : auditLogFields.getDatabase(),
+              TSFileConfig.STRING_CHARSET),
+          new Binary(
+              auditLogFields.getSqlString() == null ? "null" : auditLogFields.getSqlString(),
+              TSFileConfig.STRING_CHARSET),
+          new Binary(log == null ? "null" : log, TSFileConfig.STRING_CHARSET)
+        });
+    insertStatement.setDataTypes(
+        new TSDataType[] {
+          TSDataType.TEXT,
+          TSDataType.TEXT,
+          TSDataType.TEXT,
+          TSDataType.TEXT,
+          TSDataType.TEXT,
+          TSDataType.TEXT,
+          TSDataType.BOOLEAN,
+          TSDataType.TEXT,
+          TSDataType.TEXT,
+          TSDataType.TEXT
+        });
+    return insertStatement;
+  }
+
+  public static void log(AuditLogFields auditLogFields, String log) throws IllegalPathException {
+    if (!tableViewisInitialized) {
+      Statement statement =
+          StatementGenerator.createStatement(
+              "CREATE DATABASE " + SystemConstant.AUDIT_DATABASE, ZoneId.systemDefault());
+      ExecutionResult result =
+          COORDINATOR.executeForTreeModel(
+              statement,
+              SESSION_MANAGER.requestQueryId(),
+              sessionInfo,
+              "",
+              ClusterPartitionFetcher.getInstance(),
+              SCHEMA_FETCHER);
+      if (result.status.getCode() == TSStatusCode.SUCCESS_STATUS.getStatusCode()
+          || result.status.getCode() == TSStatusCode.DATABASE_ALREADY_EXISTS.getStatusCode()) {
+        statement =
+            StatementGenerator.createStatement(
+                "CREATE VIEW __view_system.audit_log (\n"
+                    + "    dn_id STRING TAG,\n"
+                    + "    user_name STRING TAG,\n"
+                    + "    cli_hostname STRING FIELD,\n"
+                    + "    audit_event_type INT32 FIELD,\n"
+                    + "    operation_type INT32 FIELD,\n"
+                    + "    privilege_type INT32 FIELD,\n"
+                    + "    privilege_level INT32 FIELD,\n"
+                    + "    result BOOLEAN FIELD,\n"
+                    + "    database STRING FIELD,\n"
+                    + "    log STRING FIELD\n"
+                    + ") AS root.__audit.log",
+                ZoneId.systemDefault());
+        SqlParser relationSqlParser = new SqlParser();
+        IClientSession session = SESSION_MANAGER.getCurrSession();
+        Metadata metadata = LocalExecutionPlanner.getInstance().metadata;
+        COORDINATOR.executeForTableModel(
+            statement,
+            relationSqlParser,
+            session,
+            SESSION_MANAGER.requestQueryId(),
+            SESSION_MANAGER.getSessionInfoOfTableModel(session),
+            "",
+            metadata,
+            config.getQueryTimeoutThreshold());
+        tableViewisInitialized = true;
```

Comment:
Plz inquire the concurrent safety of ur current implementation. The `create database` is safe since the ConfigNode will reject repeated database creation, while I'm not sure about the `create view`. Because the `create view` could be executed simultaneously across cluster's DataNodes.